benchmarks/CMakeLists.txt

```diff
@@ -10,16 +21,28 @@ function(ConfigureBench CMAKE_BENCH_NAME CMAKE_BENCH_SRC)
                    ${CMAKE_BENCH_SRC}
                    "${CMAKE_CURRENT_SOURCE_DIR}/synchronization/synchronization.cpp")
     set_target_properties(${CMAKE_BENCH_NAME} PROPERTIES POSITION_INDEPENDENT_CODE ON)
-    target_link_libraries(${CMAKE_BENCH_NAME} benchmark pthread rmm )
+    target_include_directories(${CMAKE_BENCH_NAME} PRIVATE
+                               "${THRUST_INCLUDE_DIR}"
+                               "${GBENCH_INCLUDE_DIR}"
+                               "${CMAKE_CUDA_TOOLKIT_INCLUDE_DIRECTORIES}"
+                               "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}>"
+                               "$<BUILD_INTERFACE:${CMAKE_SOURCE_DIR}/include>"
+                               "$<INSTALL_INTERFACE:include>")
+    target_link_directories(${CMAKE_BENCH_NAME} PRIVATE "${GBENCH_LIBRARY_DIR}")
+    
+    target_link_libraries(${CMAKE_BENCH_NAME} benchmark pthread
+                          spdlog::spdlog_header_only ${CUDART_LIBRARY})
+
     set_target_properties(${CMAKE_BENCH_NAME} PROPERTIES
-                            RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/gbenchmarks")
+                          RUNTIME_OUTPUT_DIRECTORY "${CMAKE_BINARY_DIR}/gbenchmarks")
+    
+    target_compile_definitions(${CMAKE_BENCH_NAME} PUBLIC SPDLOG_ACTIVE_LEVEL=SPDLOG_LEVEL_${LOGGING_LEVEL})
+    
+    if(USE_NVTX)
+      target_compile_definitions(${CMAKE_BENCH_NAME} PUBLIC NVTX_DISABLE)
+    endif(USE_NVTX)
```

Comment:
Same thing, users of rmm basically shouldn't even notice that it became header-only, not having to add all that stuff themselves.

Reply:
removed